app/code/Magento/PageCache/Model/App/FrontController/BuiltinPlugin.php

```diff
@@ -89,9 +89,12 @@ public function aroundDispatch(
      */
     protected function addDebugHeaders(ResponseHttp $result)
     {
-        $cacheControl = $result->getHeader('Cache-Control')->getFieldValue();
-        $this->addDebugHeader($result, 'X-Magento-Cache-Control', $cacheControl);
-        $this->addDebugHeader($result, 'X-Magento-Cache-Debug', 'MISS', true);
+        $cacheControlHeader = $result->getHeader('Cache-Control');
+        if ($cacheControlHeader instanceof \Zend\Http\Header\HeaderInterface) {
+            $cacheControl = $cacheControlHeader->getFieldValue();
+            $this->addDebugHeader($result, 'X-Magento-Cache-Control', $cacheControl);
+            $this->addDebugHeader($result, 'X-Magento-Cache-Debug', 'MISS', true);
```

Comment:
Because in that case there is no `Cache-Control` header is set. If we set any value to `X-Magento-Cache-Control`, it may confuse developers that value is set from Magento, unless we can put a very clear message to identify that fact. That is my thinking.